src/ray/object_manager/plasma/store.cc

```diff
@@ -355,9 +356,17 @@ void PlasmaStore::DisconnectClient(const std::shared_ptr<Client> &client) {
   create_request_queue_.RemoveDisconnectedClientRequests(client);
 }
 
-Status PlasmaStore::ProcessMessage(const std::shared_ptr<Client> &client,
-                                   fb::MessageType type,
-                                   const std::vector<uint8_t> &message) {
+void PlasmaStore::HandleClientConnectionError(const std::shared_ptr<Client> &client,
```

Comment:
Passing const reference of shared pointer is not a good practice.

Reply:
why's that? this avoids reference counting increments/decrements when there is no ownership transfer or storing of the ptr right?

Reply:
There're a few acceptable semantics:
- Pass a unique pointer by value: transfer ownership
- Pass shared pointer by value: increase reference count
- Pass const reference: immutable borrow semantics
- Pass mutable pointer: mutable borrow semantics
- Copy: create another object

Reply:
are you suggesting that it's better practice to dereference the shared_ptr at the callsite and pass it as a const reference?

Reply:
yes, if possible, let me know if it's hard to do and if I could help here

Reply:
the usage of pointer / reference / unique pointer / shared pointer is part of the interface which indicates borrow / ownership semantics